### ISSUE TITLE
Error "Payload is set but no content-type header specified" when calling play()

### ISSUE THREAD

**Issue by this-username-is-taken** [JMPerez/spotify-web-api-js#175]
**Describe the bug**

I'm seeing an issue where when I call `play()` without any parameters, I get an exception with the error message "Payload is set but no content-type header specified".

**To Reproduce**
This reproduces every time I call `play()`. I am using React Native and reproducing this on an Android phone. Interestingly, I'm only seeing this for `play()` but not `pause()`. I also have another project that uses React (instead of React Native) and runs in desktop Chrome and am not observing this issue.

**Additional context**
Below is the full response returned by Spotify:

```js
{
    "UNSENT": 0,
    "OPENED": 1,
    "HEADERS_RECEIVED": 2,
    "LOADING": 3,
    "DONE": 4,
    "readyState": 4,
    "status": 0,
    "timeout": 0,
    "withCredentials": true,
    "upload": {},
    "_aborted": false,
    "_hasError": true,
    "_method": "PUT",
    "_response": "Payload is set but no content-type header specified",
    "_url": "https://api.spotify.com/v1/me/player/play",
    "_timedOut": false,
    "_trackingName": "unknown",
    "_incrementalEvents": false,
    "_requestId": null,
    "_headers": {
        "authorization": "Bearer BQD1k6iCo2ITQdQTzX6VfGmYwGdeK7rAoAufwi4YDUzJI2EFnENC-q_Z1x2QljnT0TooLJbHUQMn_vKYv5zwig_4ZRWBdboZpFRZ8FuocHBcLMEp-ib-7Ia1r3c8Q87H3Jwu30Hr8ZUr2fOlnx77_w9c2bLIQAAcQW8"
    },
    "_responseType": "",
    "_sent": true,
    "_lowerCaseResponseHeaders": {},
    "_subscriptions": []
}
```

After some trial and error, I discovered that if I remove the line `postData: postData` from `request.Data` in `play()`, things will work fine:
![image](https://user-images.githubusercontent.com/2354636/84739013-b7e1bd80-af5f-11ea-9224-db2b5e38e927.png)

Despite the workaround, I'm still interested in getting to the bottom of this, and hopefully fix it in a non-hacky way if possible.

**Comment by this-username-is-taken** [comment]
Seems like a better fix is to add `contentType: 'application/json'` to `requestData`.

**Comment by JMPerez** [comment]
I think we could add a check on `_performRequest()` that adds `contentType: 'application/json'` if the type of `params` is `object` and is not null. That's more elegant than adding the contentType to a few functions.

@this-username-is-taken I'll prepare a change, and maybe you can try it out.

**Comment by this-username-is-taken** [comment]
Thanks @JMPerez , I can confirm that this is now fixed!

**Comment by JMPerez** [comment]
@this-username-is-taken awesome!

**Comment by quentin-sommer** [comment]
Hi,
I don't think the version including this fix (1.5.1) is released: [npm](https://www.npmjs.com/package/spotify-web-api-js)
@JMPerez could you release it? I'm having the same issue on react-native

**Comment by JMPerez** [comment]
Hi @quentin-sommer, you're right! I have just published 1.5.1

**Comment by quentin-sommer** [comment]
Thanks a lot
